src/frontend/js/components/DashboardItem/Order/DashboardItemOrder.spec.tsx

```diff
@@ -0,0 +1,772 @@
+import { Order, OrderState, Product } from 'types/Joanie';
+import { DATETIME_FORMAT, DEFAULT_DATE_FORMAT } from 'utils/useDateFormat';
+import { createTestQueryClient } from 'utils/test/createTestQueryClient';
+import { SessionProvider } from 'data/SessionProvider';
+import { resolveAll } from 'utils/resolveAll';
+import { DashboardPaths } from 'utils/routers/dashboard';
+import { confirm } from 'utils/indirection/window';
+import { CourseRun, Priority } from 'types';
+import { sleep } from 'utils/sleep';
+import { expectBannerError } from 'utils/test/expectBannerError';
+import { DashboardTest } from '../../Dashboard/DashboardTest';
+import Mock = jest.Mock;
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockFactories
+    .ContextFactory({
+      authentication: { backend: 'fonzie', endpoint: 'https://demo.endpoint' },
+      joanie_backend: { endpoint: 'https://joanie.endpoint' },
+    })
+    .generate(),
+}));
+
+jest.mock('utils/indirection/window', () => ({
+  confirm: jest.fn(() => true),
+}));
+
+describe('<DashboardItemOrder/>', () => {
+  const wrapper = ({ children }: PropsWithChildren) => {
+    const router = createMemoryRouter([{ index: true, element: children }]);
+    return (
+      <QueryClientProvider client={createTestQueryClient({ user: true })}>
+        <IntlProvider locale="en">
+          <SessionProvider>
+            <RouterProvider router={router} />
+          </SessionProvider>
+        </IntlProvider>
+      </QueryClientProvider>
+    );
+  };
+  const WrapperWithDashboard = (route: string) => {
+    return (
+      <QueryClientProvider client={createTestQueryClient({ user: true })}>
+        <IntlProvider locale="en">
+          <SessionProvider>
+            <DashboardTest initialRoute={route} />
+          </SessionProvider>
+        </IntlProvider>
+      </QueryClientProvider>
+    );
+  };
+
+  beforeEach(() => {
+    fetchMock.get('https://joanie.endpoint/api/v1.0/addresses/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/credit-cards/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/orders/', []);
+    fetchMock.get('https://joanie.endpoint/api/v1.0/enrollments/', []);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    fetchMock.restore();
+  });
+
+  const mockProduct = (order: Order) => {
+    const product: Product = ProductFactory.generate();
+    product.id = order.product;
+    fetchMock.get('https://joanie.endpoint/api/v1.0/products/' + product.id + '/', product);
+    return product;
+  };
+
+  /**
+   * Global
+   */
+
+  it('crashes if no course is provided', async () => {
+    const order: Order = { ...OrderFactory.generate(), course: undefined };
+    // Hide console.error ( https://stackoverflow.com/questions/66328549/testing-an-error-thrown-by-a-react-component-using-testing-library-and-jest )
+    jest.spyOn(console, 'error').mockImplementation(() => {});
```

Comment:
There is a `noop` util for that
```suggestion
    jest.spyOn(console, 'error').mockImplementation(noop);
```